e2e_samples/fabric_dataops_sample/docs/adr_naming_convention.md

```diff
@@ -0,0 +1,79 @@
+# Naming convention for Fabric E2E sample
+
+This document describes the naming convention for the Fabric E2E sample. It includes the naming convention for the following:
+
+- Git Folders
+- Scripts (python, shell, powershell etc.)
+- Configuration files (.yml, .json, .cfg etc.)
+- Azure/Fabric resources
+
+## Status
+
+Proposed
+
+## Context
+
+As the Fabric E2E sample is a complex solution with multiple components, it is important to have a consistent naming convention for the various components. This will help in better organization and management of the solution. It would also help multiple developers to work on the solution without any confusion.
+
+## Proposal details
+
+### GitHub files naming convention
+
+| Item Type | Convention | Additional Details | Example |
+| --- | --- | --- | --- |
+| Folders | snake_case | Use lowercase and "_" | `fabric_dataops_sample`, `dataops_pipeline` |
+| Markdown files | snake-case | Use lowercase and "_" | "README.md", `adr_naming_convention.md` |
+| Python scripts | snake_case | Use lowercase and "_" | `setup_fabric_environment.py` |
+| Shell scripts | snake_case | Use lowercase and "_" | `deploy_infrastructure.sh` |
+| PowerShell scripts | PascalCase | Use "-" and "verb-noun" structure | `Set-DeploymentPipelines.ps1` |
+| Images (diagrams, screenshots etc.) | kebab-case | Use lowercase and "-" | `graph-api-permission.png`, `fabric-architecture.drawio`|
+| YAML files | kebab-case | Use lowercase and "-" | `deploy-infra.yml` |
+| JSON files | kebab-case | Use lowercase and "-" | `fabric-config.json` |
+| Configuration files | kebab-case | Use lowercase and "-" | `dataops-config.cfg` |
+| Wheel files | [pep-427](https://peps.python.org/pep-0427/) | {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl | `azure_monitor_opentelemetry-1.6.1-py3-none-any.whl` |
+
+### Variables/Functions naming convention
+
+| Variable Type | Convention | Additional Details | Example |
+| --- | --- | --- | --- |
+| Environment variables | UPPER_SNAKE_CASE | Use uppercase and "_" | `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET` |
+| Shell script variables to read Terraform output | `tf_`snake_case | Prefix with "tf_" | `tf_storage_account_name` |
+| Shell/Python script variables (in general) | snake_case | Use lowercase and "_" | `storage_account_name` |
+| PowerShell script variables | `$`camelCase |  | `$storageAccountName` |
+| Shell/Python Functions | snake_case | Use lowercase and "_" | `get_storage_account_name` |
+| PowerShell Functions | PascalCase | Use "Verb-Noun" structure | `Get-StorageAccountName` |
+| Terraform variables | snake_case | Use lowercase and "-" | `environment_name` |
+
+### Azure/Fabric resource naming convention
```

Comment:
So you are really making this specifically for the e2e sample for Fabric. Shouldn't we have a Universal convention for the entire repo? Meaning - for a Fabric  Single Tech sample for Fabric you should be using also this convention - but you are creating a silo on the E2E sample. I would have a unique convention for the entire repo. Maybe Fabric notebook would be nb-fb -... and a databricks notebooks would be nb-db-.... so we can be broader with our convention